### ISSUE TITLE
[Proposal] Adding a (score) test for (non-)proportional odds

### ISSUE THREAD

**Issue by Generalized** [AnestisTouloumis/multgee#6]
Dear @AnestisTouloumis 
Thank you very much for your awesome, extraordinary package! You did an excellent work for which I'm grateful!

Let me ask, please, would it be possible, in future, t**o add also the test of non-proportionality of hazards**? Even if of limited usefulness, this is often requested by statistical reviewers... Currently only the repolr package has a score test for it.

I tried also a different approach with the Brant test (Wald's approach), which supports only with `MASS::polr` and doesn't account for clustered/repeated/correlated responses. But anyway, I fitted a longitudinal analysis as if there was no within-subject correlation (="independence" residual covariance structure) just to test the PO with it. Most of the time results were comparable, but not always. Also, sometimes polr converged while repolr didn't and vice versa, so I couldn't compare them in all possible cases. This kind of work seems a bit "slippery". 

These tests are described in: 

> Stiger TR, Barnhart HX, Williamson JM. Testing proportionality in the proportional odds model fitted with GEE. Statistics in Medicine

(I can share with you the PDF if you find it useful)

Also here:

> Liu, Anqi et al. “On testing proportional odds assumptions for proportional odds models.” General psychiatry vol. 36,3 e101048. 27 Jun. 2023, doi:10.1136/gpsych-2023-101048
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10410795/

**Comment by adrianolszewski** [comment]
Indeed, this would be an important addition, Professor @AnestisTouloumis!
